amaranth/vendor/xilinx.py

```diff
@@ -343,14 +343,20 @@ def vendor_toolchain(self):
 
     _symbiflow_part_map = {
         "xc7a35ticsg324-1L": "xc7a35tcsg324-1", # Arty-A7
+        "xc7a15tcpg236-1": "xc7a15tcpg236-1", # Cmod-A7-15
+        "xc7a35tcpg236-1": "xc7a35tcpg236-1", # Cmod-A7-35
     }
 
     _symbiflow_bitstream_device_map = {
             "xc7a35ticsg324-1L": "artix7",
+            "xc7a15tcpg236-1": "artix7",
+            "xc7a35tcpg236-1": "artix7",
```

Comment:
I'm quite sure this map could be replaced with an `if`/`else` chain, since a prefix of `xc7a` implies the device will be Artix-7.

Reply:
This looks a bit like a mess (applies to all three comments here):

The [Arty A7 reference manual](https://digilent.com/reference/programmable-logic/arty-a7/reference-manual) mentions a part called `xc7a35ticsg324-1L`, but a speed grade of '1L' does not exist in the [Artix 7 family](https://docs.xilinx.com/v/u/en-US/ds181_Artix_7_Data_Sheet), only '1LI'. The actual Arty A7 I have on hand *does not even have* a speed grade marking on the package, so I can neither confirm nor disprove.

To me it looks as if Digilent has managed to swap around some letters in the part name, while the proper part name should most probably be `xc7a35t-l1csg324i`, which is what you can order from the part vendors.

Symbiflow on the other hand has a naming scheme that is still different and requires the `_symbiflow_part_map` to map to their internal names.

The part names mentioned on the [CMOD A7 reference manual](https://digilent.com/reference/programmable-logic/cmod-a7/reference-manual) and the [CMOD S7 reference manual](https://digilent.com/reference/programmable-logic/cmod-s7/reference-manual) correctly refer to existing Xilinx parts.

Perhaps it is best - at least for new additions - to use the proper part names to avoid any more confusion.

To answer this particular comment: The `_symbiflow_bitstream_device_map` can be replaced by matching the prefix of the part name, yes. I'll do just that.

Reply:
Hmm, of course the maps rely on whatever is returned from the `_part` property. The format returned from there is also not what one would expect from the vendor datasheet.

Reply:
Oh: of course Vivado has a naming scheme different from both the datasheets and the device markings. The `_part` property returns something that Vivado understands. To use Symbiflow, the maps exist to transform into Symbiflow part names.

Reply:
This can be joined with the `_xray_family` property. Can't decide which style is nicer. `symbiflow_bitstream_device` is slightly more generic, while `_xray_family` is more concise. Also I prefer to raise an exception here instead of relying on useful error messages in the called tool.